errors.go

```diff
@@ -0,0 +1,164 @@
+	Detail string `json:"detail"`
+
+	// Status is the HTTP status code applicable to this problem, expressed as a string value.
+	Status string `json:"status,omitempty"`
+
+	// Code is an application-specific error code, expressed as a string value.
+	Code string `json:"code,omitempty"`
+
+	// TODO: (thedodd): add this when we have an internal model to use.
+	// Links is an array of link objects containing hyper-links to further details about
+	// this particular occurrence of the problem.
+	// Links []*Link `json:"links,omitempty"`
+
+	// TODO: (thedodd): add this when we have an internal model to use.
+	// Source is an object containing references to the source of the error.
+	// Source *Source `json:"source,omitempty"`
+
+	// Meta is an object containing non-standard meta-information about the error.
+	Meta *map[string]string `json:"meta,omitempty"`
+}
+
+// Error implements the `Error` interface.
+func (e *ErrorObject) Error() string {
+	return fmt.Sprintf("Error: %s %s\n", e.Title, e.Detail)
+}
+
+// GetID implements the `ErrorIDCompatible` interface.
+func (e *ErrorObject) GetID() string { return e.ID }
+
+// GetTitle implements the `ErrorTitleCompatible` interface.
+func (e *ErrorObject) GetTitle() string { return e.Title }
+
+// GetDetail implements the `ErrorDetailCompatible` interface.
+func (e *ErrorObject) GetDetail() string { return e.Detail }
+
+// GetStatus implements the `ErrorStatusCompatible` interface.
+func (e *ErrorObject) GetStatus() string { return e.Status }
+
+// GetCode implements the `ErrorCodeCompatible` interface.
+func (e *ErrorObject) GetCode() string { return e.Code }
+
+// GetMeta implements the `ErrorMetaCompatible` interface.
+func (e *ErrorObject) GetMeta() *map[string]string { return e.Meta }
+
+/////////////////////////////////////////////
+// JSON API Error Compatibility Interfaces //
+/////////////////////////////////////////////
+
+// ErrorIDCompatible is the interface needed for exposing the `id` field of a JSON API compatible error.
+type ErrorIDCompatible interface {
+	GetID() string
+}
+
+// ErrorTitleCompatible is the interface needed for exposing the `title` field of a JSON API compatible error.
+type ErrorTitleCompatible interface {
+	GetTitle() string
+}
+
+// ErrorDetailCompatible is the interface needed for exposing the `detail` field of a JSON API compatible error.
+type ErrorDetailCompatible interface {
+	GetDetail() string
+}
+
+// ErrorStatusCompatible is the interface needed for exposing the `status` field of a JSON API compatible error.
+type ErrorStatusCompatible interface {
+	GetStatus() string
+}
+
+// ErrorCodeCompatible is the interface needed for exposing the `code` field of a JSON API compatible error.
+type ErrorCodeCompatible interface {
+	GetCode() string
+}
+
+// ErrorMetaCompatible is the interface needed for exposing the `meta` field of a JSON API compatible error.
+type ErrorMetaCompatible interface {
+	GetMeta() *map[string]string
```

Comment:
I think this should be `*map[string]interface{}` as meta can usually be anything~